gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -169,4 +178,10 @@ private QueryNode checkAndAddSortAndPaginationNode(
         executionContext.getEntitiesRequest().getOffset(),
         orderByExpressions);
   }
+
+  private boolean isSingleSourceAndSame(Set<String> firstSource, Set<String> secondSource) {
```

Comment:
`firstSource.size() == 1 && firstSource.equals(secondSource)`?

Reply:
This is the same as the one in ExecutionVisitor?

Reply:
fixed.